server/routes/user.js

```diff
@@ -112,6 +116,68 @@ router.post(
   }
 );
 
+//Get all notifications of a user Id
+router.get("/getAllNotifications/:id", async (req, res) => {
+  try {
+    const { id } = req.params;
+
+    //Through the UserNotification Model from below, we'll be able to obtain the
+    //notification_id that is only present on the particular user
+    const userNotification = await UserNotificationModel.findAll({
+      where: { user_id: id },
+    });
+
+    const promiseArr = [];
+
+    //Get all the notifications for each of the values in the array
+    userNotification.forEach((userNotif) => {
+      promiseArr.push(
+        NotificationModel.findOne({ where: { id: userNotif.notification_id } })
+      );
+    });
```

Comment:
This could be another good situation for [`.map()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).